executor/src/witgen/mod.rs

```diff
@@ -163,6 +216,7 @@ impl<'a, 'b, T: FieldElement> WitnessGenerator<'a, 'b, T> {
             .analyzed
             .committed_polys_in_source_order()
             .into_iter()
+            .filter(|(symbol, _)| symbol.stage.unwrap_or_default() <= self.phase.into())
```

Comment:
Does it make sense to have this distinction between `phase` and `stage`?

Reply:
I was wondering the same, probably not! I guess we could rename stage -> phase?

Reply:
stage comes from pil, phase comes from halo.

Or a certain level, I have the impression that "stage" conveys that all stages are uniform to some degree, while with "phase", all the phases have different characters.